src/svg/Painter.ts

```diff
@@ -340,11 +340,13 @@ class SVGPainter implements PainterBase {
         this._oldVNode = null;
     }
     toDataURL(base64?: boolean) {
-        const str = encodeURIComponent(this.renderToString());
+        let str = encodeURIComponent(this.renderToString());
         const prefix = 'data:image/svg+xml;';
-        return base64
-            ? prefix + 'base64,' + window.btoa(unescape(str))
-            : prefix + 'charset=UTF-8,' + str;
+        if (base64) {
+            str = encodeBase64(str);
+            return str ? prefix + 'base64,' + str : null;
```

Comment:
I'm not sure which one is better if base64 encode is not supported. Still return the string before encode or return null.

Seems return null will be better because developers can check the return easily. If they still wan't the string before encode. They can call this method again without base64.

Reply:
> Seems return null will be better because developers can check the return easily

Yes. If it returns the unencoded string with the `base64` option, the developer may be confused.

Reply:
Agree, we can give an error log.